pydantic/_internal/_decorators.py

```diff
@@ -218,3 +256,127 @@ def in_ipython() -> bool:
         return False
     else:  # pragma: no cover
         return True
+
+
+class OnlyValueValidator(Protocol):
+    def __call__(self, __value: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithValues(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any]) -> Any:
+        ...
+
+
+class V1ValidatorWithValuesKwOnly(Protocol):
+    def __call__(self, __value: Any, *, values: dict[str, Any]) -> Any:
+        ...
+
+
+class V1ValidatorWithKwargs(Protocol):
+    def __call__(self, __value: Any, **kwargs: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithKwargsAndValue(Protocol):
+    def __call__(self, __value: Any, values: dict[str, Any], **kwargs: Any) -> Any:
+        ...
+
+
+class V1ValidatorWithValuesAndKwargs(Protocol):
```

Comment:
I noticed all of these Protocols seem to be defined in _both_ `pydantic._internal._decorators` and `pydantic.decorators` — any chance we can unify and get them defined in just one place?

Reply:
Unfortunately, no. The ones in `pydantic.decorators` add a `cls` argument since we do the magic classmethod conversion.

Which does bring up the topic: @dmontagu you had said you backed the idea of removing this magic and instead requiring users to add an `@classmethod` themselves.

I do think maybe we could add more overloads and say "if you give me a function with `cls` as the first argument I will give you back a classmethod, otherwise I will give you back the same thing I got.

Reply:
The main reason I don't like the idea of converting methods into classmethods using our own decorator is that I think it won't play nice with editors (I've found myself annoyed about that in the past).

I'm not sure whether that is relevant to the point about a function-with-cls-as-first-argument or not, I think I could be convinced about any behavior with free-standing functions.

Reply:
I don't like it either, but we need it for V1 compatibility. I organized the types so that we can easily deprecate that approach in the future. I didn't add a deprecation warning here but we can. The adding overloads point was about maybe making the situation with IDEs better. FWIW though Pylance seems to be very happy with what this PR is currently doing.